docs/proposals/remote-artifacts.md

```diff
@@ -0,0 +1,267 @@
+container image which contains this software will be specified via environment variable, and when
+not informed will use a default value.
+
+Therefore, the operator needs to generate `wget` commands with arguments in order to download the
+artifacts specified.
+
+The step generated by the operator will look like the following:
+
+```yml
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: artifact-download
+spec:
+  steps:
+    - name: artifact-download
+      image: busybox:latest
+      command:
+        - /bin/bash
+      args:
+        - -c
+        - >
+          wget --output="/workspace/source/classpath/main.jar" https://nexus.company.com/app/main.jar && \
+          chown 1000:1001 /workspace/source/classpath/main.jar && \
+          chmod 0644 /workspace/source/classpath/main.jar
+```
+
+## Example Use-Case
+
+Using [Shipwright's proposed logos](https://github.com/shipwright-io/build/issues/325) as example,
+let's assume we are building a [TypeScript application](https://github.com/otaviof/typescript-ex)
+which will use the project logo, and we would like to create two different builds, one with the
+default project logo, and another with the alternative.
+
+By using remote artifacts, we can keep the separation of project source code and assets, and we
+can describe those resources as:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: BuildArtifact
+metadata:
+  name: ship-logo
+spec:
+  artifacts:
+    - name: ship-logo
+      Url: https://user-images.githubusercontent.com/2587818/92114986-69bfb600-edfa-11ea-820e-96cdb1014f58.png
+      path: $(workspace)/assets/images/shipwright-logo.png
+```
+
+And, the alternative logo:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: BuildArtifact
+metadata:
+  name: axes-logo
+spec:
+  artifacts:
+    - name: axes-logo
+      url: https://user-images.githubusercontent.com/2587818/92100668-c1ebbd80-ede4-11ea-9e8a-7379c3875ea0.png
+      path: $(workspace)/assets/images/shipwright-logo.png
+```
+
+Then, we can create the `Build` resource, as per:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: typescript-ex
+spec:
+  strategy:
+    name: buildpacks-v3
+    kind: ClusterBuildStrategy
+  source:
+    url: https://github.com/otaviof/typescript-ex.git
+  artifacts:
+    - ship-logo
+  output:
+    image: quay.io/otaviof/typescript-ex:latest
+```
+
+Now, we can create two `BuildRun` resources, overwriting `.spec.artifacts` to compose the alternative build. For instance:
+
+
+```
+---
+apiVersion: build.dev/v1alpha1
+kind: BuildRun
+metadata:
+  name: typescript-ex
+spec:
+  buildRef:
+    name: typescript-ex
+```
+
+Also:
+
+```
+---
+apiVersion: build.dev/v1alpha1
+kind: BuildRun
+metadata:
+  name: typescript-ex-alternative-logo
+spec:
+  buildRef:
+    name: typescript-ex
+  output:
+    image: quay.io/otaviof/typescript-ex:alternative
+  artifacts:
+    - axes-logo
```

Comment:
This is where making the `name` required and having strategic merge patching becomes critical. Does having an override list here mean "replace all" or "replace only these"